doc/build/sysbuild/index.rst

```diff
@@ -648,3 +693,285 @@ each image that is part of a project. Alternatively, there are
+
+Exclusion takes precedence over inclusion when evaluating the contents of each
+group. For example, recall that the image ``my_sample`` has been added to both
+``my_group`` and ``other_group``, but it is filtered out from ``another_group``.
+Furthermore, trying to add that image back to the group by naïvely extending it:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     another_group
+     INCLUDE my_sample
+   )
+
+would not work, because the prior ``EXCLUDE_FROM`` directive still applies with
+higher priority. If you do need this to work, then you should consider defining
+yet another group:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     yet_another_group
+     INCLUDE_FROM another_group
+     INCLUDE my_sample
+   )
+
+In this way, ``yet_another_group`` can successfully include ``my_sample``,
+even though ``another_group`` excludes it. This is because group contents are
+evaluated in isolation, so by the time ``another_group`` is merged in, its own
+inclusions and exclusions will already have been squashed. The same goes for
+``my_group`` and ``other_group``, which need to be resolved like this first.
+
+Adding dependencies among sysbuild groups
+=========================================
+
+Just like for individual Zephyr applications, you can add configuration or
+flashing dependencies for groups of these applications.
+See :ref:`sysbuild_zephyr_application_dependencies` for more information.
+
+Here is an example of group-wise dependencies:
+
+.. code-block:: cmake
+
+   sysbuild_add_dependencies(GROUP CONFIGURE my_group group_a group_b)
+   sysbuild_add_dependencies(GROUP FLASH     my_group group_a group_b)
+
+This will ensure that every image in ``group_a`` and ``group_b`` will be
+configured and flashed before every image in ``my_group``. The ordering within
+the groups themselves is not determined here. Moreover, groups are not treated
+as units; once other image-wise and group-wise dependencies are factored in,
+the images belonging to one group are not expected to stick together in the
+sorted order.
+
+Note that ``my_group`` is allowed to overlap with ``group_a`` or ``group_b``.
+If two groups overlap and one group depends on the other, then the images that
+the two groups have in common will be placed in the middle of the sorted order,
+relative to the images unique to either group.
+
+To help illustrate what this means, consider a different example:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Group(
+     group_a
+     INCLUDE i_1 i_2 i_3 i_4 i_5 i_6
+   )
+   ExternalZephyrProject_Group(
+     group_b
+     INCLUDE i_1 i_2 i_3 i_7 i_8 i_9
+   )
+   sysbuild_add_dependencies(GROUP FLASH group_a group_b)
+
+The flashing order represented by this dependency would be something like this:
+
+.. code-block:: none
+
+      ┌───────────────────────────── unique to group_b
+      │                     ┌─────── unique to group_a
+     i_7        i_1        i_4
+   { i_8 } -> { i_2 } -> { i_5 }
```

Comment:
A neat addition for the future would be a ninja command that prints out a graphic like this for a configuration showing the flash/configuration order

Reply:
Definitely. So far, I only added a partial printout for when you get a cyclic dependency error.